src/WebJobs.Extensions.DurableTask/DurableTaskOptions.cs

```diff
@@ -14,20 +14,29 @@ namespace Microsoft.Azure.WebJobs.Extensions.DurableTask
     public class DurableTaskOptions
     {
         /// <summary>
-        /// The default task hub name to use when not explicitly configured.
-        /// </summary>
-        internal const string DefaultHubName = "DurableFunctionsHub";
-
-        /// <summary>
-        /// Gets or sets default task hub name to be used by all <see cref="DurableOrchestrationClient"/>,
+        /// Gets or sets task hub name to be used by all <see cref="DurableOrchestrationClient"/>,
         /// <see cref="DurableOrchestrationContext"/>, and <see cref="DurableActivityContext"/> instances.
         /// </summary>
         /// <remarks>
         /// A task hub is a logical grouping of storage resources. Alternate task hub names can be used to isolate
         /// multiple Durable Functions applications from each other, even if they are using the same storage backend.
         /// </remarks>
         /// <value>The name of the default task hub.</value>
-        public string HubName { get; set; } = DefaultHubName;
+        public string HubName { get; set; }
+
+        /// <summary>
+        /// Sets the default value of the task hub name to the function application name. Should only be used if 
+        /// no value for HubName was already provided.
+        /// </summary>
+        /// <remarks>
+        /// This method utilizes the WEBSITE_SITE_NAME environment variables that is set by the Azure App Service infrastructure.
+        /// If this environment variable is not present (i.e. running locally or on non-App Service infrastructure), we default
+        /// to localhost.
+        /// </remarks>
+        internal void SetDefaultHubName(INameResolver nameResolver)
+        {
+            this.HubName = nameResolver.Resolve("WEBSITE_SITE_NAME") ?? "localhost";

```

Comment:
I considered using `WEBSITE_IIS_SITE_NAME` to allow differentiation between the site and its slots, but decided against it for a few reasons:

1. It prepends "~1" to the sitename, which could be confusing.
2. It doesn't exist on Linux.
3. The suffix for slot names isn't deterministic to the best of my knowledge. It may not even be consistent between site start-ups or slot swaps.  This could lead to some confusing behavior for the customers.

1 could be easily fixed, and in the case of 2 we could just default to `WEBSITE_SITE_NAME` if `WEBSITE_IIS_SITE_NAME` doesn't exist. The third point is my largest concern. It might be best just to insist that customers set their own TaskHub names if they want to use slots, and provide best practices for them.

Reply:
What about a combination of `WEBSITE_SITE_NAME` and `WEBSITE_SLOT_NAME`? And maybe if it's the production slot, we leave that suffix out and just make it `WEBSITE_SITE_NAME`?

We tried the guidance approach already, but unfortunately too many people don't read this far into the docs and end up getting into a bad situation, which is why I want to have reasonable defaults.

Reply:
As per our offline discussion, decided to remove the default for now. We can always add one later without it being a breaking change if we find a good value to use.